identifierModule.cpp

```diff
@@ -0,0 +1,32 @@
+// Takes pID, testval(int) and addresses list as input and returns the address stored if  testval is a pointer.
+#include "ProcessManager.h"
+using namespace std;
+
+void ProcessManager :: identifierFunction(long unsigned int pid, int testval, int** add, int size) {
+		// int testval = (int)strtol(test, NULL,16);
+		int i =0;
+		int count=0;
+		// cout <<size <<endl;
+		while( i < size ){
+    	// if(*(*(add+i)+0) < *(*(add+i)+1)) {	
+			if( *(*(add+i)+0) < testval && testval < *(*(add+i)+1)) {
+				cout << "Its there! \n";
+				printf("Address Stored: %p \n", test);
+				count +=1;
+			}
+		// }
+		// else {
+			if( *(*(add+i)+1) < testval && testval < *(*(add+i)+0)) {
+				cout << "Its there \n";
+				printf("Address Stored: %p \n", test);
+				count+=1;
+			}
+		// }
+		i++;
+	}
+	if(count==0) {
+		cout << "it isnt here!";
+	}
```

Comment:
This is the function which checks if the given memory address is a pointer right?
Then why would it take add as an input and also pid.
It should take only one input which would be the memory address which is to be checked if it is a pointer or not.

Reply:
Ok, I'll try to change that.